### PR TITLE
Update downloads to mv3

### PR DESCRIPTION
**Pull request by Shubham-Rasal**
This PR aims to upgrade the existing download extension samples to MV3.  Here I have migrated the downloads manager and download links samples. 
If this PR is appropriate, I would love to continue to update others as well

### CODE REVIEW

**Comment by patrickkettner** [comment]
@oliverdunk would you be able to review?

File: api-samples/downloads/download_links/send_links.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  if (((i > 0) && (links[i] == links[i - 1])) ||
+      (links[i] == '') ||
+      (kBadPrefix == links[i].toLowerCase().substr(0, kBadPrefix.length))) {
+    links.splice(i, 1);
+  } else {
+    ++i;
+  }
+}
+
+console.log(links)
```
Comment (oliverdunk) [inline api-samples/downloads/download_links/send_links.js:35]: We weren't previously logging here, so I think we should probably remove this?

File: api-samples/downloads/download_links/manifest.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "Download Selected Links",
+  "description": "Select links on a page and download them.",
+  "version": "0.3",
+  "minimum_chrome_version": "16.0.884",
+  "permissions": [
+    "downloads",
+    "activeTab",
```
Comment (oliverdunk) [inline api-samples/downloads/download_links/manifest.json:8]: Thanks for moving to activeTab here!

File: api-samples/downloads/download_manager/background.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  } else if (options.anyRecentlyCompleted) {
+    drawCompleteBadge(ctx);
+  }
+  return canvas;
+}
+
+function maybeOpen(id) {
+  var openWhenComplete = [];
+  try {
+    openWhenComplete = JSON.parse(localStorage.openWhenComplete);
```
Comment (oliverdunk) [inline api-samples/downloads/download_manager/background.js:143]: localStorage isn't available in MV3. We'll need to update this before we can test the extension.

**Review by oliverdunk — CHANGES_REQUESTED** [review]
Thanks so much for taking a look at this!

I left some initial thoughts - in both cases the extensions aren't working right now so it would be great to fix that. Maybe you had some extra changes that you didn't commit?

It would also be great to run the linter (`npm run lint:fix`). Note that this will change `var` to `let` which will break the `download_links` extension when you open the popup more than once, since it will try to re-declare variables. We'll need to look at fixing that.

**Comment by Shubham-Rasal** [comment]
Thanks for the feedback. On it.

**Comment by jpmedley** [comment]
Does either of you have time to add READMEs to these?

**Comment by Shubham-Rasal** [comment]
Yeah I will add them, once the code is ready.

**Comment by Shubham-Rasal** [comment]
Sorry, I won't be able to work on this. You can go ahead with your plan

**Pull request closed without merging**